src/AreaChart/AreaSeries/Line.tsx

```diff
@@ -77,69 +83,55 @@ export interface LineProps extends PropFunctionTypes {
   hasArea: boolean;
 }
 
-interface LineState {
-  pathLength?: number;
-}
-
-export class Line extends PureComponent<LineProps, LineState> {
-  static defaultProps: Partial<LineProps> = {
-    showZeroStroke: true,
-    strokeWidth: 3
-  };
-
-  state: LineState = {};
-  ghostPathRef = createRef<SVGPathElement>();
-
-  componentDidMount() {
-    if (this.ghostPathRef.current) {
-      this.setState({
-        pathLength: this.ghostPathRef.current!.getTotalLength()
-      });
+export const Line: FC<Partial<LineProps>> = ({
+  showZeroStroke = true,
+  strokeWidth = 3,
+  interpolation,
+  data,
+  xScale,
+  yScale,
+  width,
+  hasArea,
+  animated,
+  color,
+  index,
+  style,
+  className
+}) => {
+  const [pathLength, setPathLength] = React.useState<number>(null);
+  const ghostPathRef = createRef<SVGPathElement>();
```

Comment:
```suggestion
  const ghostPathRef = createRef<SVGPathElement | null>(null);
```

Reply:
actually it is not needed since it is the actual creation of the reference(the createRef doesn't accept any params as well).

Reply:
Oh silly - this should actually be `useRef` - I mis-read this.